### PR TITLE
Add null_value support to geo_point type

### LINKED ISSUE #12998: Ability to set default geo_point value
Spoke with @nknize about the possibility of setting a default geo_point lat/lon value, and he said it would be possible to add this feature.

### PR DESCRIPTION
**Pull request by imotov**
Adds support for null_value attribute to the geo_point types.

Closes #12998

### CODE REVIEW

**Comment by jpountz** [comment]
I'd like to understand what the use-case is for such a feature. I can understand how a default value equal to "n/a" can help on `keyword` fields, but I can't think of a use-case for a default location.

**Comment by imotov** [comment]
@kurtado, since you opened the original issue, could you comment on it?

**Comment by kurtado** [comment]
Sure, from what I recall, we had a customer in training who wished to set a default, exactly like this. I can't remember the exact use case, but a `null_value` works fine.

File: server/src/main/java/org/elasticsearch/common/geo/GeoUtils.java [hunk 1]
```diff
@@ -351,6 +357,32 @@ public static GeoPoint parseGeoPoint(XContentParser parser, GeoPoint point) thro
...
+     * Object: has to contain either lat and lon or geohash fields
+     * <p>
+     * String: expected to be in "latitude, longitude" format or a geohash
+     * <p>
+     * Array: two or more elements, the first element is longitude, the second is latitude, the rest is ignored if ignoreZValue is true
+     */
+    public static GeoPoint parseGeoPoint(Object value, final boolean ignoreZValue) throws ElasticsearchParseException {
+        try {
+            XContentBuilder content = JsonXContent.contentBuilder();
+            content.value(value);
```
Comment (jpountz) [inline server/src/main/java/org/elasticsearch/common/geo/GeoUtils.java]: Should we put the value under a field in a top-level object? My reasoning is that I wouldn't be surprised that we start failing producing malformed json documents at some point, which would be the case here if you give a top-level string which is not enclosed in an object?

**Review by jpountz — APPROVED** [review]
Fair enough. I doubt this will be used much, but at least this is consistent with other fields.

File: server/src/main/java/org/elasticsearch/common/geo/GeoUtils.java (hunk 1 above)
Comment (imotov) [reply server/src/main/java/org/elasticsearch/common/geo/GeoUtils.java]: Yes, I was thinking about it as well. Why the current solution is a bit faster it indeed feels somewhat hacky. I will add a top level object.

**Pull request merged**